lib/locale_flash/flash.rb

```diff
@@ -10,11 +10,31 @@ def initialize(args={})
       @message    = args[:message]
     end
 
-    def self.from_params(type, params={})
-      if params.is_a?(Hash) && params[:controller] && params[:action]
-        new(params.merge(:type => type))
-      else
-        new(:type => type, :message => params.to_s)
+    class << self
+      def from_params(type, params={})
```

Comment:
```
=> :a_private_method
[4] pry(main)> ClassName.a_private_method
=> nil
[5] pry(main)> ClassName.new.a_private_method
NoMethodError: private method `a_private_method' called for #<ClassName:0x00000808882470>
from (pry):20:in `__pry__'
```

If I recall correctly, when you use `def self.something`, since the
receiver is explicit, it's not affected by `public`, `private`, or
`protected`.

But I eventually found a solution, seems to work with
```
class Foo
  private_class_method def self.a_private_method
  end
end
```